### PR TITLE
[1748] Updates to "Record the form error messages to GA"

### PR DESCRIPTION
**Pull request by markmead**
### Context

Updates from product review.

### Changes proposed in this pull request

- Updated error messages to have `data-ga-event-form`
- Added `data-error-message` to errors
- Changed `form-publish` to `form-submitted`

### CODE REVIEW

File: app/views/layouts/_flash.html.erb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <div class="grid-row">
   <div class="column-two-thirds">
     <% flash.each do |key, value| %>
-      <div class="govuk-<%= key %>-summary" role="alert" aria-labelledby="<%= key %>-summary-heading" tabindex="-1" data-module="error-summary">
-        <h3 class="govuk-<%= key %>-summary__title" id="<%= key %>-summary-heading"><%= value %></h3>
+      <div class="govuk-<%= key %>-summary" role="alert" aria-labelledby="<%= key %>-summary-heading" tabindex="-1" data-module="error-summary" data-ga-event-form="error">
+        <h3 class="govuk-<%= key %>-summary__title" id="<%= key %>-summary-heading" data-error-message="<%= value %>"><%= value %></h3>
```
Comment (fofr) [inline app/views/layouts/_flash.html.erb]: We should only include `data-error-message` if the key is `error`.

File: app/views/layouts/_flash.html.erb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <div class="grid-row">
   <div class="column-two-thirds">
     <% flash.each do |key, value| %>
-      <div class="govuk-<%= key %>-summary" role="alert" aria-labelledby="<%= key %>-summary-heading" tabindex="-1" data-module="error-summary">
-        <h3 class="govuk-<%= key %>-summary__title" id="<%= key %>-summary-heading"><%= value %></h3>
+      <div class="govuk-<%= key %>-summary" role="alert" aria-labelledby="<%= key %>-summary-heading" tabindex="-1" data-module="error-summary" data-ga-event-form="error">
```
Comment (fofr) [inline app/views/layouts/_flash.html.erb]: Maybe:
`data-ga-event-form="error"
```suggestion
      <div class="govuk-<%= key %>-summary" role="alert" aria-labelledby="<%= key %>-summary-heading" tabindex="-1" data-module="error-summary" data-ga-event-form="<%= key %>">
```

Comment (markmead) [reply app/views/layouts/_flash.html.erb]: This change resolve the above?

**Comment by markmead** [comment]
[Record each success and error message to Google Analytics as a GA Event](https://trello.com/c/vS2xhvvx/1748-record-each-success-and-error-message-to-google-analytics-as-a-ga-event)

File: app/views/layouts/_flash.html.erb (hunk 2 above)
Comment (fofr) [reply app/views/layouts/_flash.html.erb]: No, the message will still show in the wrong context.

**Pull request merged**